File: BlinkID/android/src/main/java/com/microblink/flutter/MicroblinkScannerViewFactory.java
package com.microblink.flutter;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.Map;

import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
import io.flutter.plugin.common.BinaryMessenger;
import io.flutter.plugin.common.StandardMessageCodec;
import io.flutter.plugin.platform.PlatformView;
import io.flutter.plugin.platform.PlatformViewFactory;

class MicroblinkScannerViewFactory extends PlatformViewFactory {
    private final BinaryMessenger messenger;
    private final ActivityPluginBinding activityPluginBinding;

    MicroblinkScannerViewFactory(BinaryMessenger messenger,
                                 ActivityPluginBinding activityPluginBinding) {
        super(StandardMessageCodec.INSTANCE);

        this.messenger = messenger;
        this.activityPluginBinding = activityPluginBinding;
    }

    @NonNull
    @Override
    public PlatformView create(@NonNull Context context, int id, @Nullable Object args) {
        return new MicroblinkScannerView(
                context,
                id,
                new MicroblinkCreationParams((Map<String, Object>) args),
                messenger,
                activityPluginBinding
        );
    }
}

File: BlinkID/android/src/main/kotlin/com/microblink/blinkid/flutter/MicroblinkScannerViewFactory.kt
package com.microblink.blinkid.flutter

import android.content.Context
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

internal class MicroblinkScannerViewFactory(
    private val messenger: BinaryMessenger,
    private val activityPluginBinding: ActivityPluginBinding
) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {
    override fun create(context: Context, id: Int, args: Any?): PlatformView {
        return MicroblinkScannerView(
            context,
            id,
            MicroblinkCreationParams((args as Map<String?, Any?>?)!!),
            messenger,
            activityPluginBinding
        )
    }
}